modules/local/nf-core-modified/sourmash/compare/main.nf

```diff
@@ -0,0 +1,48 @@
+process SOURMASH_COMPARE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::sourmash=4.6.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.6.1--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.6.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signatures)
+    val(seqtype)
+    path file_list // optional file
+    val save_numpy_matrix
+    val save_csv
+
+    output:
+    tuple val(meta), path("*comp.npy.labels.txt"), path("*comp.npy"), optional:true, emit: matrix
+    path "*comp.csv"       , optional:true, emit: csv
+    path "versions.yml"    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args     ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}.${seqtype}"
+    def comp   = save_numpy_matrix ? "--output ${seqtype}.comp.npy"  : ''
+    def csv    = save_csv          ? "--csv ${seqtype}.comp.csv" : ''
+    if ( !save_numpy_matrix && !save_csv ) error "Supply either save_numpy_matrix, save_csv, or both or no output will be created"
+    def ffile = file_list ? "--from-file ${file_list}" : ''
+    def sigs = signatures ? "${signatures.join(' ')}" : ''
+    if ( !file_list && !signatures ) error "Supply either signatures, file_list, or both"
+    """
+    sourmash compare \\
+        $args \\
+        --processes $task.cpus \\
+        ${comp} \\
+        ${csv} \\
+        ${ffile} \\
+        ${sigs}
```

Comment:
you might want to add `--ignore-abundance` as we expect wildly different abundances between reads and assemblies, and we dont want that to downweight similarity metrics